### PR TITLE
feat: athena sql stage

### PR DESCRIPTION
**Pull request by malachi-constant**
- #141  (Implementation)
- Updating cdk version in dependencies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/pipelines/stage.ts [hunk 1]
```diff
@@ -42,16 +51,90 @@ export abstract class DataStage extends Stage {
...
     return this;
   }
 }
 
+export abstract class StateMachineStage extends DataStage {
+  readonly targets?: events.IRuleTarget[];
+  readonly eventPattern?: events.EventPattern;
+
+  public stateMachine?: sfn.StateMachine;
+  public stateMachineInput?: { [key: string]: any };
```
Comment (LeonLuttenberger) [inline src/pipelines/stage.ts:73]: These two don't need to be optional, do they? A state machine will always be created. As far the input is concerned, if we don't get it in the props, we would just set it to `{}`.

File: src/pipelines/stage.ts [hunk 2]
```diff
@@ -42,16 +51,90 @@ export abstract class DataStage extends Stage {
...
+        var statement,
+          _pj_c = 0,
+          _pj_a = props.additionalRolePolicyStatements,
+          _pj_b = _pj_a.length;
+        _pj_c < _pj_b;
+        _pj_c += 1
+      ) {
+        statement = _pj_a[_pj_c];
+
+        this.stateMachine.addToRolePolicy(statement);
```
Comment (LeonLuttenberger) [inline src/pipelines/stage.ts:109]: This seems like a very complicated and hard to read way of writing a for-loop. You can simplify it as:
```typescript
if (props.additionalRolePolicyStatements) {
  props.additionalRolePolicyStatements.forEach(s => {
    this.stateMachine.addToRolePolicy(s);
  });
}
```

**Review by LeonLuttenberger — CHANGES_REQUESTED** [review]

File: src/pipelines/stage.ts (hunk 1 above)
Comment (malachi-constant) [reply src/pipelines/stage.ts:73]: `stateMachine` should be required, changing that, but `stateMachineInput` should allow for undefined/null value.

Comment (LeonLuttenberger) [reply src/pipelines/stage.ts:73]: When a null value is given for `stateMachineInput`, doesn't this mean we should pass an empty object to the state machine? I'm not sure what exactly happens to `RuleInput.fromObject(stateMachineInput)` when `stateMachineInput` is null.

Comment (malachi-constant) [reply src/pipelines/stage.ts:73]: I follow you now `{}` vs `undefined`

**Review by LeonLuttenberger — APPROVED** [review]

**Pull request merged**